### DISCUSSION TITLE
How to have long equations scrollable on smaller screens?

### DISCUSSION THREAD

**Discussion by dense-set** [quarto-dev/quarto-cli#9248] [Q&A] [answered]
### Description

I am coming across instances where I need to write long equations on my webpage. This works fine for big screens, but is gnarly for smaller ones. For instance, see the following screenshots of a [blogpost](https://dense-set.github.io/blog/steele-csmc/ch-1.html) that I recently authored:

<img width="417" alt="Screenshot 2024-04-02 at 12 48 02 AM" src="https://github.com/quarto-dev/quarto-cli/assets/102873841/8aa8aa65-b091-4942-a378-dbc8bb1130ec">

<img width="314" alt="Screenshot 2024-04-02 at 12 47 56 AM" src="https://github.com/quarto-dev/quarto-cli/assets/102873841/30f83740-ee42-4687-852b-e467b3550cd6">

I am aware that MathJax provides a way to control the spillages on small screens by making them "scrollable", as is done on MathSE like so:

https://github.com/quarto-dev/quarto-cli/assets/102873841/75c4c393-e5ec-4595-8aa2-e770b9481055

**Question:** How to enable this on Quarto webpages? I tried adding the following to CSS (as suggested in [here](https://stackoverflow.com/a/61942403/11953295)) but to no avail.
```css
.math.display {
    overflow-x: auto;
    overflow-y: hidden;
}
```

**Comment by mcanouil** [comment]
I understand you're looking for assistance with HTML/CSS, particularly regarding screen width rules. While this topic falls outside the scope of Quarto support, I'd be happy to guide you towards resources that can help.

Learning about CSS and its application in controlling layout for different screen sizes can be quite useful. A great place to start is online tutorials that focus on CSS. For instance, you might find the CSS reference on media queries quite useful for understanding responsive design principles (https://www.w3schools.com/cssref/css3_pr_mediaquery.php). Additionally, the developer tools in web browsers can be incredibly insightful for experimenting with and debugging your CSS code.

I hope this helps

Related topics:
- https://github.com/quarto-dev/quarto-cli/discussions/5550
- https://github.com/quarto-dev/quarto-cli/discussions/2170

**Reply by dense-set** [reply, marked as answer]
Thanks for such a friendly response! Having so many tools, each doing so many things, I often guess wrongly their associations. But with you guys here so patiently listening to me and always generous with your tips, I do find solutions evetually. This is one of those instances.

I realized that the math that MathJax injects is in `<span>` rather than `<div>`, and my original solution thus failed, for `overflow-x` can be specified only for block level elements. It turns out that what I must have been targeting is `mth-container[display="ture"]`. And that worked as a charm!

**Reply by phineas-pta** [reply]
there's a typo in answer if anyone wondering: it should be `mjx-container[display="true"]`
